projects/graph/graph.py

```diff
@@ -11,45 +11,142 @@ def add_vertex(self, vertex):
+                return
+            print(f'DFS recursive vertex {vertex}')
+            visited.add(vertex)
+            for neighbor in self.vertices[vertex]:
+                if neighbor not in visited:
+                    inner_recursive(neighbor)
+
+        # print(f'dft recursive')
+        # visited = set()
+        # def inner_recursive(vertex):
+        #     visited.add(vertex)
+        #     
+        #     print(f'Recursive vertex = {vertex} {visited}')
+        #     for neighbor in self.vertices[vertex]:
+        #         if neighbor not in visited:
+        #             inner_recursive(neighbor)
+
+        #     print(f' Nothing printed from {visited} because this visited is not part of recursive loop.')
+  
+        inner_recursive(starting_vertex)
+
     def bfs(self, starting_vertex, destination_vertex):
         """
         Return a list containing the shortest path from
         starting_vertex to destination_vertex in
         breath-first order.
         """
-        pass  # TODO
+        # Create an empty set to store visited nodes
+        visited = set()
+        # Create an empty Queue and enqueue A PATH TO the starting vertex
+        q = Queue()
+        path = [starting_vertex]
+        q.enqueue(path)
+        # While the queue is not empty...
+        while q.size() > 0 :
+            # Dequeue the first PATH
+            path = q.dequeue()
+            # GRAB THE VERTEX FROM THE END OF THE PATH
+            v = path[-1]
+            # IF VERTEX = TARGET, RETURN PATH
+            if v == destination_vertex :
+                return(f'BFS shortest path {path}')
+            # If that vertex has not been visited...
+            if v not in visited:
+                # Mark it as visited
+                visited.add(v)
+                # Then add A PATH TO all of its neighbors to the back of the queue
+                for neighbor in self.vertices[v]:
+                    # Copy the path
+                    path_copy = path.copy()
+                    # Append neighbor to the back of the copy
+                    path_copy.append(neighbor)
+                    # Enqueue copy
+                    q.enqueue(path_copy)
+
     def dfs(self, starting_vertex, destination_vertex):
         """
         Return a list containing a path from
         starting_vertex to destination_vertex in
         depth-first order.
         """
-        pass  # TODO
+        visited = set()
+        s = Stack()
+        path = [starting_vertex]
+        s.push(path)
+        while s.size() > 0:
+            path = s.pop()
+            v = path[-1]
+            if v == destination_vertex :
+                return(f'DFS shortest path {path}')
+            
+            if v not in visited:
+                visited.add(v)
+                for neighbor in self.vertices[v]:
+                    path_copy = path.copy()
+                    path_copy.append(neighbor)
+                    s.push(path_copy)
 
```

Comment:
very nice; here again I think you would want to have the destination check inside the v not in visited before you add